### PR TITLE
New References – Reference section on the application review page

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

Candidates no longer need to wait for their references come back before they can submit their application. The review page needs to be changed to only display the referee details and accept the submission if there is a minimum of 2 references.

## Changes proposed in this pull request
|Before|After|
|---|---|
|image|image|

## Guidance to review

1. See the cards each scenario and test on your local machine (you need to use the cycle switcher to after apply 1 deadline and enable the feature flag)

## Link to Trello card

https://trello.com/c/bJrKTSr9/423-references-new-references-section-on-the-application-review-page

### CODE REVIEW

File: config/locales/candidate_interface/review_application.yml [hunk 1]
```diff
@@ -69,6 +69,13 @@ en:
     references:
       incomplete: Add %{minimum_references} referees to your application
       complete_section: Add your references
+    new_references:
+      incomplete: References section not marked as complete
+      complete_section: Complete references section
+      one_reference_only: You need to add a second reference
+      add_more_references: Add references
```
Comment (frankieroberto) [inline config/locales/candidate_interface/review_application.yml:76]: I think we should say "Add another reference" here, for consistency with the button text in the section itself.

```suggestion
      add_more_references: Add another reference
```

File: app/components/candidate_interface/incomplete_new_references_section_component.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module CandidateInterface
+  class IncompleteNewReferencesSectionComponent < IncompleteSectionComponent
+    include ViewHelper
+
+    def message; end
```
Comment (frankieroberto) [inline app/components/candidate_interface/incomplete_new_references_section_component.rb:5]: This null-method is a bit confusing. Is it needed to override an inherited method? Maybe add a comment?

**Review by frankieroberto — APPROVED** [review]
I tested this in the review app and it works like a charm, with all the routing when changing or deleting references from the application review working as expected, and all the right messages showing if zero or 1 references have been added, or if the section hasn't been marked as complete. 👍 

Screenshot 2022-08-05 at 14 22 49

File: config/locales/candidate_interface/review_application.yml (hunk 1 above)
Comment (JR-G) [reply config/locales/candidate_interface/review_application.yml:76]: 👍

File: app/components/candidate_interface/new_references_review_component.rb [hunk 3]
```diff
@@ -1,13 +1,45 @@
...
+    def incomplete_section_params
+      {
+        section: :references_selected,
+        section_path: candidate_interface_new_references_review_path,
+        error: @missing_error,
+      }.merge(incomplete_section_content)
+    end
+
+    def incomplete_section_content
+      if @references.size > 1 && @application_form.references_completed.blank?
```
Comment (carlosmartinez) [inline app/components/candidate_interface/new_references_review_component.rb:28]: activerecord helper: `@references.many?`

File: app/components/candidate_interface/new_references_review_component.rb [hunk 4]
```diff
@@ -1,13 +1,45 @@
...
+        text = t('review_application.new_references.incomplete')
+        link_text = t('review_application.new_references.complete_section')
+      elsif @references.size == 1
```
Comment (carlosmartinez) [inline app/components/candidate_interface/new_references_review_component.rb:31]: activesupport helper: `@references.one?`

File: app/components/candidate_interface/new_references_review_component.rb [hunk 5]
```diff
@@ -151,5 +183,13 @@ def feedback_status_label(reference)
         colour: t("candidate_reference_colours.#{reference.feedback_status}"),
       )
     end
+
+    def return_to_params
+      if @return_to_application_review
+        { 'return_to' => 'application-review' }
+      else
+        { 'return_to' => 'review' }
```
Comment (carlosmartinez) [inline app/components/candidate_interface/new_references_review_component.rb:191]: i note `:review` has changed to `'review'` here; i guess rails isn't bothered?

File: config/locales/candidate_interface/references.yml [hunk 6]
```diff
@@ -89,6 +89,7 @@ en:
         more_than_two: More than 2 references have been given
         need_two: You need to have received at least 2 references before marking this section complete
         incorrect_number_selected: You need to have exactly 2 references selected before submitting your application
+        incorrect_number: You need to have 2 references minimum before submitting your application
```
Comment (carlosmartinez) [inline config/locales/candidate_interface/references.yml:92]: I think "at least 2 references" is nicer phrasing than "2 references minimum"

File: config/locales/candidate_interface/review_application.yml [hunk 7]
```diff
@@ -69,6 +69,13 @@ en:
     references:
       incomplete: Add %{minimum_references} referees to your application
       complete_section: Add your references
+    new_references:
+      incomplete: References section not marked as complete
+      complete_section: Complete references section
+      one_reference_only: You need to add a second reference
+      add_more_references: Add another references
```
Comment (carlosmartinez) [inline config/locales/candidate_interface/review_application.yml:76]: Should be either "Add another reference" or "Add more references"

**Review by carlosmartinez — APPROVED** [review]

File: app/components/candidate_interface/new_references_review_component.rb (hunk 5 above)
Comment (JR-G) [reply app/components/candidate_interface/new_references_review_component.rb:191]: apparently not!

**Pull request merged**